src/pytorch_lightning/strategies/ddp_spawn.py

```diff
@@ -263,30 +278,37 @@ def reduce(self, tensor, group: Optional[Any] = None, reduce_op: Union[ReduceOp,
             tensor = sync_ddp_if_available(tensor, group, reduce_op=reduce_op)
         return tensor
 
-    def training_step(self, *args, **kwargs) -> STEP_OUTPUT:
+    def training_step(self, *args: Any, **kwargs: Any) -> STEP_OUTPUT:
+        assert self.model is not None
         with self.precision_plugin.train_step_context():
             return self.model(*args, **kwargs)
 
-    def validation_step(self, *args, **kwargs) -> Optional[STEP_OUTPUT]:
+    def validation_step(self, *args: Any, **kwargs: Any) -> Optional[STEP_OUTPUT]:
         with self.precision_plugin.val_step_context():
+            assert self.lightning_module is not None
+            assert self.model is not None
             if self.lightning_module.trainer.state.fn == TrainerFn.FITTING:
                 # used when calling `trainer.fit`
                 return self.model(*args, **kwargs)
             else:
                 # used when calling `trainer.validate`
-                return self.model.validation_step(*args, **kwargs)
+                return self.model.validation_step(*args, **kwargs)  # type: ignore[operator]
 
-    def test_step(self, *args, **kwargs) -> Optional[STEP_OUTPUT]:
+    def test_step(self, *args: Any, **kwargs: Any) -> Optional[STEP_OUTPUT]:
+        assert self.model is not None
         with self.precision_plugin.test_step_context():
-            return self.model.test_step(*args, **kwargs)
+            return self.model.test_step(*args, **kwargs)  # type: ignore[operator]
 
-    def predict_step(self, *args, **kwargs) -> STEP_OUTPUT:
+    def predict_step(self, *args: Any, **kwargs: Any) -> STEP_OUTPUT:
+        assert self.model is not None
         with self.precision_plugin.predict_step_context():
-            return self.model.predict_step(*args, **kwargs)
+            return self.model.predict_step(*args, **kwargs)  # type: ignore[operator]
 
-    def post_training_step(self):
+    def post_training_step(self) -> None:
+        assert self.lightning_module is not None
         if not self.lightning_module.automatic_optimization:
-            self.model.require_backward_grad_sync = True
+            assert self.model is not None
+            self.model.require_backward_grad_sync = Tensor([True])
```

Comment:
why is this becoming a tensor?

Reply:
Blind guess is that the author added it after getting the error reported in https://github.com/pytorch/pytorch/issues/81462. I suggest ignoring

Reply:
That's right. `mypy` expects `self.model.require_backward_grad_sync` to have type `Union[Tensor, Module]`. I will change the value back to `True` and `ignore` the error.

```suggestion
            self.model.require_backward_grad_sync = True  # type: ignore[assignment]
```